### PR TITLE
[RA1 Ch01] Add Exceptions

### LINKED ISSUE #1927: [RA1 Ch01] Create Exception section and add SR-IOV Exception
Pursuant to RM decision to eliminate Exceptions tracking from the RM add a section on Exceptions (with subsections on Technology and Version Exceptions -- see [9.4.2 Exception Types](https://github.com/cntt-n/CNTT/blob/master/doc/gov/chapters/chapter09.md#9.2.2)) and add the SR-IOV exception.

Currently, in RM Appendix A, the following technology exception exists:

## A.3 Exception List
As Part of the [Transition Plan](https://github.com/cntt-n/CNTT/blob/master/doc/gov/chapters/chapter09.md#9.2) described in the adoption strategy, following table explains the exceptions allowed in this RM release. The list of Exceptions described here are considered to be against CNTT principles and will be removed in future releases as soon as an alternative technology that is aligned with CNTT principles develops and matures.

Ref | Type | Name | Description
-- | -- | -- | --
rm.exc.001 | Technology | SRIOV | This exception allows workloads (VNF/CNF) to rely on SR-IOV over PCI-PassThrough technology.

### PR DESCRIPTION
**Pull request by pgoyal01**
Fixes #1927 

Added content for Exceptions from RM with modifications

### CODE REVIEW

File: doc/ref_arch/openstack/chapters/chapter01.md [hunk 1]
```diff
@@ -55,6 +56,37 @@ OpenStack considers the following Four Opens essential for success:
...
+
+The following sub-sections list the exceptions to the CNTT principles and shall be updated whenever technology choices, versions and requirements change. The Exceptions have an associted period of validity and this period shall include time for transitioning.
+
+#### 1.4.1.1 Technology Exceptions
+
+The list of Technology Exceptions will be updated or removed when alternative technologies aligned with CNTT principles develop and mature.
+
+| Ref |	Type |	Name |	Description | Valid Until |
+|-----|------|------|-------------|-------------|
+| ra1.exc.tec.001 |	Technology |	SR-IOV |	This exception allows workloads to use SR-IOV over PCI-PassThrough technology. | TBD |
```
Comment (tomkivlin) [inline doc/ref_arch/openstack/chapters/chapter01.md:72]: I think having a "rationale" and "implication" field in the table would help explain why the exception is in place.

File: doc/ref_arch/openstack/chapters/chapter01.md [hunk 2]
```diff
@@ -55,6 +56,37 @@ OpenStack considers the following Four Opens essential for success:
...
+
+#### 1.4.1.2 Version Exceptions
```
Comment (tomkivlin) [inline doc/ref_arch/openstack/chapters/chapter01.md:74]: I'm not sure I understand this section - what is the exception "for"?

File: doc/ref_arch/openstack/chapters/chapter01.md [hunk 3]
```diff
@@ -55,6 +56,37 @@ OpenStack considers the following Four Opens essential for success:
...
+
+The list of Version Exceptions will be updated as and when alternative versions become available.
+
+| Ref |	Type |	Name |	Description | Valid Until |
+|-----|------|------|-------------|-------------|
+| ra1.exc.ver.001 |	Version |	xxx |	xxxxxxxxxxxxxxxxxxxxxxxxxxxx. |
+
+
+#### 1.4.1.3 Requirements Exceptions
```
Comment (tomkivlin) [inline doc/ref_arch/openstack/chapters/chapter01.md:83]: Also not quite sure about this - is this where we would document where the requirement is an exception of the CNTT/RM spec?  Or where RA1 spec is an exception to the requirement?

File: doc/ref_arch/openstack/chapters/chapter01.md (hunk 2 above)
Comment (pgoyal01) [reply doc/ref_arch/openstack/chapters/chapter01.md:74]: @tomkivlin Version Exceptions are for software versions, API versions, etc. For example, while we are using OpenStack Train, for a given service we could limit use of API to an earlier version. The full Cyborg API v2 is available on Ussuri and only partial set on Train and so we could specify that if using Cyborg utilise only API v1 not API v2.

File: doc/ref_arch/openstack/chapters/chapter01.md (hunk 3 above)
Comment (pgoyal01) [reply doc/ref_arch/openstack/chapters/chapter01.md:83]: @tomkivlin the requirement is an exception of the CNTT/RM spec, and also if an RA1 spec is being diluted/waived

File: doc/ref_arch/openstack/chapters/chapter01.md (hunk 2 above)
Comment (tomkivlin) [reply doc/ref_arch/openstack/chapters/chapter01.md:74]: Ah ok understood.

File: doc/ref_arch/openstack/chapters/chapter01.md (hunk 1 above)
Comment (pgoyal01) [reply doc/ref_arch/openstack/chapters/chapter01.md:72]: @tomkivlin Great idea. Thanks

File: doc/ref_arch/openstack/chapters/chapter01.md [hunk 4]
```diff
@@ -55,6 +56,37 @@ OpenStack considers the following Four Opens essential for success:
 
 This OpenStack Reference Architecture is organised around the three major NFVI resource types as core services of compute, storage and networking, and a set of shared services of identity management, image management, graphical user interface, orchestration engine, etc. 
 
+<a name="1.4.1"></a>
+### 1.4.1 Exceptions
+
+CNTT specifies certain policies and [principles](https://github.com/cntt-n/CNTT/blob/master/doc/common/chapter00.md#2.0) but technology, performance and other considerations makes conformance to these primciples and policies impractical. CNTT specifies how to handle [non-conforming technologies](https://github.com/cntt-n/CNTT/blob/master/doc/common/policies.md#cntt-policies-for-managing-non-conforming-technologies). In general, non-coformance with policies is handled through a set of exceptions (please also see [Exception Types](https://github.com/cntt-n/CNTT/blob/master/doc/gov/chapters/chapter09.md#942-exception-types)).
```
Comment (iangardner22) [inline doc/ref_arch/openstack/chapters/chapter01.md:62]: Typo "primcipes"

Also the main thing that jumps out about this sentence is why does CNTT specify the policies if "technology, performance and other considerations makes conformance impractical"...    are saying in the future they may be practical or?

File: doc/ref_arch/openstack/chapters/chapter01.md [hunk 5]
```diff
@@ -55,6 +56,37 @@ OpenStack considers the following Four Opens essential for success:
...
 
+<a name="1.4.1"></a>
+### 1.4.1 Exceptions
+
+CNTT specifies certain policies and [principles](https://github.com/cntt-n/CNTT/blob/master/doc/common/chapter00.md#2.0) but technology, performance and other considerations makes conformance to these primciples and policies impractical. CNTT specifies how to handle [non-conforming technologies](https://github.com/cntt-n/CNTT/blob/master/doc/common/policies.md#cntt-policies-for-managing-non-conforming-technologies). In general, non-coformance with policies is handled through a set of exceptions (please also see [Exception Types](https://github.com/cntt-n/CNTT/blob/master/doc/gov/chapters/chapter09.md#942-exception-types)). 
+
+The following sub-sections list the exceptions to the CNTT principles and shall be updated whenever technology choices, versions and requirements change. The Exceptions have an associted period of validity and this period shall include time for transitioning.
```
Comment (iangardner22) [inline doc/ref_arch/openstack/chapters/chapter01.md:64]: Typo "associted"

File: doc/ref_arch/openstack/chapters/chapter01.md [hunk 6]
```diff
@@ -55,6 +56,37 @@ OpenStack considers the following Four Opens essential for success:
...
+
+#### 1.4.1.1 Technology Exceptions
+
+The list of Technology Exceptions will be updated or removed when alternative technologies aligned with CNTT principles develop and mature.
+
+| Ref |	Name |	Description | Valid Until | Rationale | Implication |
+|-----|------|-------------|-------------|-----------|-------------|
+| ra1.exc.tec.001 |	SR-IOV |	This exception allows workloads to use SR-IOV over PCI-PassThrough technology. | TBD | Emulation of virtual devices for each virtual machine creates an I/O bottleneck resulting in poor performance and limits the number of virtual machines a physical server can support. SR-IOV implements virtual devices in hardware, and by avoiding the use of a switch, near maximal performance can be achieved. | Compromises virtualisation and creates dependnecy on hardware defeating CNTT and Cloud Principles. |
```
Comment (iangardner22) [inline doc/ref_arch/openstack/chapters/chapter01.md:72]: Typo "dependnecy"

File: doc/ref_arch/openstack/chapters/chapter01.md (hunk 5 above)
Comment (pgoyal01) [reply doc/ref_arch/openstack/chapters/chapter01.md]: @iangardner22 Thanks - Done

File: doc/ref_arch/openstack/chapters/chapter01.md (hunk 4 above)
Comment (pgoyal01) [reply doc/ref_arch/openstack/chapters/chapter01.md]: @iangardner22 Can you review the new language? Thanks

File: doc/ref_arch/openstack/chapters/chapter01.md (hunk 3 above)
Comment (pgoyal01) [reply doc/ref_arch/openstack/chapters/chapter01.md:83]: @tomkivlin Can you review the new language? Thanks

File: doc/ref_arch/openstack/chapters/chapter01.md (hunk 6 above)
Comment (pgoyal01) [reply doc/ref_arch/openstack/chapters/chapter01.md]: @iangardner22 Thanks - Done

**Review by tomkivlin — APPROVED** [review]

File: doc/ref_arch/openstack/chapters/chapter01.md (hunk 4 above)
Comment (iangardner22) [reply doc/ref_arch/openstack/chapters/chapter01.md]: I feel we are slightly contradicting ourselves...  how about "CNTT strives to bring the industry towards a standard approach for NFVi but some technology options available today or specific carrier constraints mean that this is not always achievable therefore an exception process exists to manage this situation."

Comment (iangardner22) [reply doc/ref_arch/openstack/chapters/chapter01.md]: might need a comma or two in my sentence sorry!

Comment (pgoyal01) [reply doc/ref_arch/openstack/chapters/chapter01.md]: @iangardner22 I had tried to leave the original content from RM the same but have now made changes in line with your suggestions -- please review. Thanks

**Review by iangardner22 — APPROVED** [review]

**Review by walterkozlowski — APPROVED** [review]

**Comment by pgoyal01** [comment]
@scottsteinbrueck Can you please approve and merge? Thanks

**Pull request merged**